enhancements/alert-on-operator-updates.md

```diff
@@ -0,0 +1,170 @@
+services) running across a Kubernetes cluster. Given that the installed Operators have the freedom to do almost anything with appropriate access privileges, there is a 
+potential threat that these actions may cause harm to the cluster, are malicious, or detrimental to a cluster upgrade. 
+
+This enhancement deals with OLM protecting the cluster from operator actions that could harm the cluster. For example: 
+- Alert is triggered when the operator deploys operands with an invalid [Pod Disruption Budget (PDB)](https://kubernetes.io/docs/tasks/run-application/configure-pdb/#specifying-a-poddisruptionbudget) that would cause an openshift upgrade issue
+- Alert is triggered when an operator tries to modify an existing on-board [Security Context Constraints (SCCs)](https://www.openshift.com/blog/understanding-service-accounts-sccs))
+ 
+
+### Goals
+
+1. Provide a generic alerting mechanism in OLM for potentially harmful operator actions
+2. Allow adding/removing a set of rules defined for various k8 objects that the operator has access to. Example objects
+ like PDB and SCCs are considered
+3. Provide configuration option to cluster admins for a permissive mode vs. enforcing mode for an operator action
+    - Able to update independently what is an alerting rule and what is an enforcing rule.
+    - When an action is deemed harmful, alert from the start and add enforcement after we get data on how many of these
+     alerts are firing.
+    - Provide option to turn off an enforcing mode
+
+### Non-Goals
+- Allow cluster admins to set the rules and externalize the configuration
+
+## Open Questions 
+
+1. What are the different ways in which a PDB & SCCs are deemed invalid by an operator action
+2. Configuration option on OLM to make for a permissive mode vs. a enforcing mode 
+ - How do we provide flexibility to cluster admin to define what rules are permitted vs enforced    
+3. Validation library is used to define what components of an Operator are deemed potentially harmful 
+	 - How do I go about finding this? Also, for these components that are deemed harmful, what is the expected action to be taken? I
+	 presume they would need a different design strategy than the one for SCCs and PDB? 
+
+## Design Details
+
+The design is split into 3 sections, each one per goal defined above. 
+
+## 1. A mechanism to alert when an invalid PDB is being applied on to a cluster (a.k.a validating PDB yaml) 
+
+Let's consider an example PDB.yaml
+
+```yaml
+apiVersion: policy/v1beta1
+kind: PodDisruptionBudget
+metadata:
+  name: zk-pdb
+spec:
+  minAvailable: 2  
+  selector:
+    matchLabels:
+      app: zookeeper
+```
+
+PodDisruptionBudget has three fields:
+
+- A label selector `.spec.selector` to specify the set of pods to which it applies. This field is required.
+- `.spec.minAvailable` which specifies the number of pods that must still be available after the eviction, even in the absence of the evicted pod.
```

Comment:
`even in the absence of the evicted pod` seems redundant after `after the eviction`